src/backend/db/database_manager.py

```diff
@@ -850,5 +866,365 @@ def delete_ticket(self, ticket_id, customer_guid):
+            logger.debug(f"Checking existence of ticket_id: {ticket_id}")
+            ticket_exists = session.execute(
+                text("SELECT 1 FROM tickets WHERE ticket_id = :ticket_id LIMIT 1"),
+                {"ticket_id": ticket_id}
+            ).fetchone()
+
+            logger.debug(f"Ticket existence check result: {ticket_exists}")
+            if not ticket_exists:
+                raise ValueError(f"Invalid ticket_id: {ticket_id} does not exist.")
+
+            # Begin transaction
+            logger.debug("Starting transaction for comment creation")
+            with session.begin_nested():
+                comment_uuid=str(uuid.uuid4())
+                # Insert into ticket_comments table
+                query = '''INSERT INTO ticket_comments (ticket_id, posted_by, comment, comment_uuid)
+                           VALUES (:ticket_id, :posted_by, :comment, :comment_uuid)'''
+                session.execute(
+                    text(query),
+                    {
+                        "ticket_id": ticket_id,
+                        "posted_by": posted_by,
+                        "comment": comment,
+                        "comment_uuid":comment_uuid
+                    },
+                )
+
+                logger.debug("Insert into ticket_comments executed")
+
+                # Fetch the generated comment_id
+                result = session.execute(
+                    text("""
+                        SELECT comment_id 
+                        FROM ticket_comments 
+                        WHERE ticket_id = :ticket_id AND comment_uuid = :comment_uuid AND posted_by = :posted_by
+                    """),
+                    {"ticket_id": ticket_id, "comment_uuid": comment_uuid, "posted_by": posted_by}
+                ).fetchone()
+
+                logger.debug(f"Fetch newly created comment_id result: {result}")
+
+                if result:
+                    comment_id = result[0]
+                    logger.debug(f"Created comment with ID: {comment_id}")
+                else:
+                    raise Exception("Failed to retrieve the newly created comment ID.")
+
+            session.commit()
+            logger.debug(f"Transaction committed successfully for comment ID {comment_id}")
+
+            return {"comment_id": comment_id, "status": "created"}
+
+        except ValueError as e:
+            logger.error(f"Validation error: {e}")
+            session.rollback()
+            raise e
+
+        except SQLAlchemyError as e:
+            logger.error(f"Database error creating comment for customer {customer_guid}: {e}")
+            session.rollback()
+            raise e
+
+        except Exception as e:
+            logger.error(f"Unexpected error while creating comment: {e}")
+            session.rollback()
+            raise e
+
+        finally:
+            logger.debug("Closing database session")
+            session.close()
+
+    def get_comment_by_id(self, comment_id, customer_guid):
```

Comment:
Change function to have four parameters def get_comment_by_id(self, comment_id, ticket_id, customer_guid):

Reply:
Modified Sir